ignite/distributed/launcher.py

```diff
@@ -315,11 +309,15 @@ def training(local_rank, config, **kwargs):
         self.logger.info("End of run")
 
     def __enter__(self) -> "Parallel":
-        if (self.backend is not None) and self._spawn_params is None:
-            idist.initialize(self.backend, init_method=self.init_method)
-            self.logger = setup_logger(__name__ + "." + self.__class__.__name__)
-            self.logger.info(f"Initialized processing group with backend: '{self.backend}'")
-
+        self.logger = setup_logger(__name__ + "." + self.__class__.__name__)
```

Comment:
Nit: lint always suggests to set up attributes inside `__init__`.

Reply:
If we use spawn, logger defined in `__init__` will be only on master process. I've preferred defining only once but outside `__init__`

Reply:
Actually, I still do not understand the purpose of moving it here.

Reply:
We always define a logger per process. Each logger is initialized wrt his rank so process with `rank=0` is initialized to write in streams.

In the actual code, we use two times `setup_logger`
* for `spawn`. It is done before process were spawned so here, we have only one logger. As you can see, in the spawn flow, this logger is not used. 
https://github.com/pytorch/ignite/blob/a2343fa8caa5a8887af691f2eb06fa77e4b0845a/ignite/distributed/launcher.py#L232
* for `torch.distributed.launch`. The setup of the logger is done right after the process group initialization, it means rank is defined, so each logger can be properly initialized wrt his rank.
https://github.com/pytorch/ignite/blob/a2343fa8caa5a8887af691f2eb06fa77e4b0845a/ignite/distributed/launcher.py#L320

The following initialization seems weird to me because it is done before the initialization of process group.
https://github.com/pytorch/ignite/blob/a2343fa8caa5a8887af691f2eb06fa77e4b0845a/ignite/distributed/launcher.py#L232
Think about it in the `torch.distributed.launch`, each logger per process is setup as `rank=0`. That is not correct. I suppose that's why a second setup is called for `torch.distributed.launch` as I mentioned above. To me, the comment shows a problem, not a fix... We should not use internally logger before initialize the process group. Or, if we use it (even in colab), it should works.

Reply:
```
self.logger = setup_logger(__name__ + "." + self.__class__.__name__, distributed_rank=0)
```
I agree that this is a hack to avoid calls like `idist.get_rank()` which will init 1 proc XLA context before calling `idist.spawn` to spawn 8 procs.

Reply:
I was also thinking if we should only rely on idist to create the logger with `setup_logger` ?
For example, we can think of using other ways to resolve the problem of creating single writing logger by using filelock things ...

Reply:
> I was also thinking if we should only rely on idist to create the logger with `setup_logger` ?
> For example, we can think of using other ways to resolve the problem of creating single writing logger by using filelock things ...

I don't know if it would work due to filesystem concerns, shared or not, permission or not, etc.